libraries/apollo-tooling/src/main/kotlin/com/apollographql/apollo3/tooling/FieldInsights.kt

```diff
@@ -32,45 +30,30 @@ object FieldInsights {
             percentile = percentile,
         )
     ).execute()
-    val data = response.data
-    return when {
-      data == null -> {
-        val cause = when (val e = response.exception!!) {
-          is ApolloHttpException -> {
-            val body = e.body?.use { it.readUtf8() } ?: ""
-            Exception("Cannot fetch field latencies: (code: ${e.statusCode})\n$body", e)
-          }
-
-          is ApolloGraphQLException -> {
-            Exception("Cannot fetch field latencies: ${e.errors.joinToString { it.message }}")
-          }
-
-          else -> {
-            Exception("Cannot fetch field latencies: ${e.message}", e)
-          }
-        }
-        FieldLatenciesResult.Error(cause = cause)
-      }
 
-      data.service == null && response.hasErrors() -> {
-        FieldLatenciesResult.Error(cause = Exception("Cannot fetch field latencies: ${response.errors!!.joinToString { it.message }}"))
-      }
+    val data = response.data
+    if (data == null) {
+      return FieldLatenciesResult.Error(cause = response.toException("Cannot fetch field latencies"))
+    }
 
-      else -> {
-        FieldLatencies(fieldLatencies = data.service?.statsWindow?.fieldLatencies?.mapNotNull {
-          val parentType = it.groupBy.parentType ?: return@mapNotNull null
-          val fieldName = it.groupBy.fieldName ?: return@mapNotNull null
-          val durationMs = it.metrics.fieldHistogram.durationMs ?: return@mapNotNull null
-          FieldLatencies.FieldLatency(
-              parentType = parentType,
-              fieldName = fieldName,
-              durationMs = durationMs
-          )
-        } ?: emptyList())
-      }
+    val service = data.service
+    if (service == null) {
+      return FieldLatenciesResult.Error(cause = Exception("Cannot find service $serviceId: ${response.errors?.joinToString { it.message }}}"))
```

Comment:
Is this 
* semantic null => no need to display `response.errors`
* error null => we could make this `@semanticNonNull`

Reply:
My hunch is semantic null but I have no proof 😅

Reply:
We can know for sure if semantic null is possible though, right? If passing a bad service id, do we get an error in the errors array?

Reply:
Just checked -> nothing in `errors` when passing a wrong service id.